volume-group-operator/controllers/persistentvolumeclaim/persistentvolumeclaim_controller.go

```diff
@@ -145,11 +150,85 @@ func (r PersistentVolumeClaimWatcher) removeVolumeFromPvcListAndPvList(logger lo
 	return nil
 }
 
+func (r PersistentVolumeClaimWatcher) addPersistentVolumeClaimToVolumeGroupObjects(
+	logger logr.Logger, pvc *corev1.PersistentVolumeClaim) error {
+	var err error
+	vgList, err := utils.GetVGList(logger, r.Client)
+	if err != nil {
+		return err
+	}
+	err = r.isPVCCanBeAddedToVG(logger, pvc, vgList)
+	if err != nil {
+		return err
+	}
+
+	for _, vg := range vgList.Items {
+		isPVCMatchesVG := false
+		if !utils.IsPVCPartOfVG(pvc, vg.Status.PVCList){
+			isPVCMatchesVG, err = utils.IsPVCMatchesVG(logger, r.Client, pvc, vg)
+			if err != nil {
+				return utils.HandleErrorMessage(logger, r.Client, &vg, err, addingPVC)
+			}
+		}
+
+		if isPVCMatchesVG {
+			err := r.addVolumeFromPvcListAndPvList(logger, pvc, &vg)
+			return utils.HandleErrorMessage(logger, r.Client, &vg, err, addingPVC)
+		}
+	}
```

Comment:
why not to put `if isPVCMatchesVG` after `utils.IsPVCMatchesVG` (inside `if !utils.IsPVCPartOfVG`) and get rid of `isPVCMatchesVG := false`

Reply:
because I didn't want to use a the function to for VGs that I won't handle anyway and this is a function that can raise an error, so the less it runs it reduces the chance to get errors

Reply:
done